internal/sqlsmith.TestGenerateParse: TestGenerateParse failed

internal/sqlsmith.TestGenerateParse [failed](https://mesolite.cluster.engflow.com/invocations/default/fc07c54f-9609-4598-86c3-a01007cdaefb?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvaW50ZXJuYWwvc3Fsc21pdGg6c3Fsc21pdGhfdGVzdA==) on master @ [b739945a4f486063d86e80a9e88afc8af7eebb89](https://github.com/cockroachdb/cockroach/commits/b739945a4f486063d86e80a9e88afc8af7eebb89):


```
        SELECT "t""ab_625"."!col""_1349" AS "col\\x69_1351" FROM (VALUES (CASE WHEN true THEN '1979-05-11':::DATE ELSE CASE WHEN true THEN '1987-03-10':::DATE ELSE '1990-06-17':::DATE END END), ('1980-07-12':::DATE), ('1996-11-10':::DATE), ('1983-05-20':::DATE), ('-infinity':::DATE)) AS "t""ab_625" ("!col""_1349"), (VALUES (('C5675265/E72DC05B':::PG_LSN, '1997-11-06 11:48:40.000357+00':::TIMESTAMPTZ, 'BOX(-1.4484976672030445 -1.5348566452583654,0.35208045534236787 1.0802008466138036)':::BOX2D, (-29700):::INT8)), ((NULL:::PG_LSN, '2004-10-23 15:09:58.000872+00':::TIMESTAMPTZ, NULL:::BOX2D, (-1385):::INT8)), (NULL), (('EC470996/D49170C1':::PG_LSN, '4714-11-24 00:00:00+00 BC':::TIMESTAMPTZ, 'BOX(-0.5240493712910396 -0.672623443341196,0.2821431404663999 0.8608380371220542)':::BOX2D, (-19341):::INT8))) AS "ta😻b _626" (col_1350) WHERE false GROUP BY "t""ab_625"."!col""_1349" LIMIT 60:::INT8;
        NULL;
        DO $$
        DECLARE
        BEGIN
        decl_97 := decl_101;
        END;
        $$;
        WHILE true LOOP
        WHILE decl_101 LOOP
        DECLARE
        "de%pcl_240" BYTES[] := ARRAY['\x28b6':::BYTES,'\x79bdbe0db5':::BYTES,'\xb9265547a9':::BYTES];
        decl_241 INT8;
        decl_242 NAME := e'\x7f2\x1f\x02Y!gj':::STRING:::NAME;
        BEGIN
        "dec%ql_92" := '01020000A0E610000004000000D0F6864A66B12940909B4886226C25C0BB6C51DA94C0F0C18E6444B364CA6040BCAF6FC2B21E4C407A513C016F2AF84100855A6E101D0DC088A2CD56A4284E405C6DEEB1EA3CE1C19E49273239BB61C0BA8ABA652596524068E6D7F60691D5C1':::GEOGRAPHY;
        FOR "lo%vop_60" IN REVERSE decl_241..decl_241 LOOP
        FOR "lo{op_61" IN REVERSE "lo%vop_60"..2259851349750390401:::INT8 LOOP
        END LOOP;
        END LOOP;
        EXIT WHEN (-7588316158100354923):::INT8::INT8 IN ();
        WITH "wIth_159" ("col""_1355") AS (SELECT * FROM (VALUES ('0.0.0.0/0':::INET), ((WITH with_158 (col_1352) AS (SELECT * FROM (VALUES ('01:40:30.073618':::TIME), ('19:09:09.43506':::TIME)) AS tab_627 (col_1352)) SELECT NULL AS "c%pol_1354" FROM (VALUES (501972488:::OID), (NULL), (2140563066:::OID)) AS tab_628 ("col,_1353") WHERE false LIMIT 1:::INT8)), ('2fe4:69e7:fe7:1ccf:8c45:41ae:21dc:5b50/33':::INET), ('37.193.137.192/26':::INET)) AS tab_629 ("col""_1355")) SELECT decl_91 AS col_1356, "decl _98" AS 😺col_1357 FROM "wIth_159" AS cte_ref_48 WHERE true LIMIT 52:::INT8;
        WITH with_160 ("c""ol_1358", "co l._1359") AS (SELECT * FROM (VALUES (inner_product('[-0.21529746,0.2917886,0.35284984,-0.72988176,1.216761,-0.5922978,0.10648398,-1.2706919,0.35868913,1.0754226,0.52124023,-0.568109,0.5444428,1.9682133,1.4796889,0.7591203,-0.5226751,-0.4546492,-0.9341307,-0.8500894,1.2068706,-0.62557626,1.5428151,-1.2687577,-0.2223071,0.49783018,0.7135745,-0.43468592,-0.13031276,-0.8837263,-0.65173775,-0.79359365,-0.07225576,3.144097,2.8391497,-0.26667148,0.1854537,0.6188545,1.16267,-1.9554727,0.92831403,0.43356174,0.5770321,-1.7755978,0.90845996,-0.59210014,-1.3926271,-0.27310252,-0.99334115,-1.3330132,0.78795475,0.11033781,1.4431716,0.4366485,0.25596043,1.1154637,-1.0939335,0.70637625,-1.4295368,0.53739125,0.8222925,-0.006153984,0.5107282,0.3188244,1.2580866,1.1020662,0.39875072,-1.7890704,1.4177428,0.3962788,-0.45473862,-0.036578026,-0.16252387,0.33403552,1.216573,-1.5732476,-0.04289119,-1.1525656,1.8592163,-0.29427186,0.89185274,1.7848227,0.76976377,-1.5110276,1.5334684,-2.0688534,-2.961813,0.30448475,1.3394259,1.2724257,-2.5815918,0.11674373,1.0567493,0.78379154,-0.17270073,-0.5558714,0.032549027,-0.3004908,0.06503957,-0.011187396,0.0713438,0.19081971,-0.9892025,0.34224826,0.6903206,0.95118773,-0.4798734,0.7171813,-0.18957205,-0.23890187,-1.2596512,1.1453484,0.7956185,0.97731453,-1.5652041,2.2298365,0.26122272,-0.7797881,0.39853498,-0.5739112,-0.71733594,-0.1358046,1.3612907,0.4140387,0.21673954,-0.042217996,1.416923,1.1149238,1.2959559,-0.03879867,0.24783278,0.5872753,-0.41551456,-0.81142485,-0.8144431,0.1249714,-2.0155592,0.46413535,-0.0030626752,0.06650532,1.1978358,-0.12828824,-1.6310096,-0.9972723,0.40509057,-0.39171073,-0.73157996,-0.42816177,-0.06780671,1.9652525,-1.8556271,-1.1653707,0.10010815,0.17633937,-0.82521445,-1.2713665,-0.7031474,1.2740288,-0.88939214,0.87920576,-0.4934377,1.5797856,0.04047686,-0.5299911,0.1724492,0.036961667,1.1162443,0.40684196,0.23298931,-1.4055715,0.22364044,-0.6118928,-0.005551949,-0.56786114,0.5042439,0.07809888,-0.20066313,0.4609201,-0.3454961,1.2818513,0.54287726,1.7595102,0.40108332,1.0409926,0.74911183,0.797973,0.7007117,0.27992955,1.7936467,-0.12162406,-0.72348964,-0.38203305,0.27684075,-1.7963026,-0.05627989,0.28319204,0.40224624,1.0927571,0.24521895,0.6128921,2.0124424,0.20286725,1.3601416,-0.64496857,-1.6690998,0.18850523,-0.8775575,0.13268319,0.34080422,-0.28707814,-1.3219335,-0.5454678,-0.63990575,-0.70812374,-1.4207634,0.98041177,0.58920354,-0.72379434,1.9095821,-2.515209,0.22283636,-0.3901979,-0.83338207,0.1465111,-1.071604,0.8309887,-0.43653366,0.35822776,-0.891365,0.51265484,0.6375719,-1.2748767,0.3351286,-0.5916561,0.023362502,0.92908674,0.29167315,0.93603146,-0.064535886,1.536315,0.22672088,1.20388,-1.0210735,0.49742424,-1.9216293,0.027985005,-0.97625536,-0.033907536,-1.0811303,0.6839543,-1.0080277,-1.3532605,-0.4306619,-1.4932352,1.7183415,-0.86156243,-1.0391434,-0.5518467,-1.5747786,-1.4093395,-0.1422474,-1.5979838,0.8262029,0.559943,0.23675571,1.0653981,-1.4802679,-0.5598177,-0.54733574,0.90693045,0.25535884,-1.1540083,-0.68023103,1.3123957,-0.02746246,1.6637685,-0.6191932,1.363851,-0.8911032,0.3324047,1.8755057,-0.10561833,-0.7042891,-0.17596336,0.3844195,-0.028677791,0.43262598,-0.32114547,0.44608736,-0.74014765,0.28480974,-0.9462149,0.28044578,0.6868217,0.23997885,0.2547376,0.67537856,0.1450249,-0.32398325,0.08721445,0.59552014,-0.7006211,0.4606777,-0.21922375,-1.071825,-0.044890527,-0.293562,1.6885167,-0.5145564,-0.52422535,-0.61443627,0.49234015,1.0088011]':::VECTOR::VECTOR, '[-0.43024367,0.35759342,0.49403298,-0.73485345,0.42356795,-0.1692512,-0.48490968,1.2204045,-0.39250413,0.7214684,0.45233995,0.82034636,1.2314174,0.53972715,1.0087903,-0.71689695,-2.744155,1.520856,-0.3258584,0.59460294,0.4777119,1.1660781,0.5881993,1.3074092,-1.4541379,-0.7668581,0.2701726,-0.27164143,-1.267562,-0.4913089,0.719346,0.32111722,-0.12710054,-1.5063092,0.11591565,0.05601244,-0.11697868,-0.9740913,-0.54624885,1.3631032,0.89647084,0.8105055,-1.6172198,-0.434417,-0.14064395,-1.1043715,0.116231516,-0.77854055,1.196949,-1.1569331,-1.5426213,0.49058223,-1.0700016,0.880006,-0.21027932]':::VECTOR::VECTOR)::FLOAT8, 19624:::INT8), ((-0.5316375189322855):::FLOAT8, 1:::INT8)) AS tab_630 ("c""ol_1358", "co l._1359")), with_161 ("co%pl_1360") AS (SELECT * FROM (VALUES ('''xHZGxXtxw'' <-> ''zNRPaB'' <-> !''IRTQToxMB'' & !''bIay'' <-> ''AlJWMPz''':::TSQUERY), ('!''cfYze''':::TSQUERY), ('''Wz''':::TSQUERY), ('''acDJlpBOgG'' | ''pbnqtCs''':::TSQUERY)) AS "\gta\fb,'_631" ("co%pl_1360")), with_162 ("!col_1361") AS (SELECT * FROM (VALUES (NULL), ('''gt'' | ''eBsIWIHlA'' & ''AKXS''':::TSQUERY), (NULL), ('''Nk'' & !''mqMBqa'' | ''ixRCRD''':::TSQUERY), ('!''GwqoERfco'' | ''Gy''':::TSQUERY)) AS tab_632 ("!col_1361")) SELECT name(NULL::CITEXT)::NAME AS col_1362 FROM with_162 AS cte_ref_49, with_161 AS cte̿_ref_50;
        END;
        "de cl""_102" := NULL;
        WITH " with_163" (col_1363) AS (SELECT * FROM (VALUES (19682:::INT8), (NULL), (int2(NULL::INTERVAL)::INT2), ((-1):::INT8), ((-4789):::INT8)) AS tab_633 (col_1363)) SELECT "d ecl'_89" AS col_1364 FROM " with_163" AS cte_ref_51, " with_163" AS "cte%0d_ref_52", " with_163" AS cte_ref_53, " with_163" AS "cte_ ref_54" LIMIT 86:::INT8 INTO STRICT "d ecl'_89";
        END LOOP;
        RETURN NULL;
        END LOOP;
        SELECT 2566175424:::REGROLE AS col_1368 FROM (VALUES ('00:00:00':::INTERVAL), ('1 day':::INTERVAL), ('00:00:00':::INTERVAL), ('-52 years -11 mons -365 days -03:36:16.142958':::INTERVAL), ('5 years 764 days 16:51:46.242449':::INTERVAL), ('3 years 2 mons 795 days 24:55:46.0893':::INTERVAL)) AS "t\\xbbab/_634" (col̅_1365), (VALUES ('bac8:ec0c:e499:ee50:6b44:b3ff:2a2:974c/58':::INET, '17:13:14.937452-05:28':::TIMETZ), ('0.0.0.0/0':::INET, '08:34:54.08642-08:01':::TIMETZ), (('230.164.204.0/29':::INET::INET & '73c2:5b65:3dfd:4e06:f898:b1dc:96cc:7719/86':::INET::INET)::INET, NULL)) AS tab_635 (col_1366, col_1367) ORDER BY tab_635.col_1367, tab_635.col_1366 DESC, "t\\xbbab/_634".col̅_1365 DESC;
        WITH "w ith_164" (col̝_1369) AS (SELECT * FROM (VALUES (()), (()), (()), (())) AS tab_636 (col̝_1369)), with_165 ("c?ol_1370") AS (SELECT * FROM (VALUES (ARRAY[5632:::INT8,9617:::INT8,(-28133):::INT8]), (ARRAY[3651:::INT8,0:::INT8,(-11871):::INT8]), (ARRAY[24535:::INT8,(-12281):::INT8,(-13769):::INT8]), (ARRAY[127:::INT8,0:::INT8,(-1):::INT8,(-31563):::INT8,29735:::INT8]), (ARRAY[(-25901):::INT8,1844:::INT8,(-20560):::INT8]), (ARRAY[8603:::INT8,(-4407):::INT8,16166:::INT8])) AS tab_637 ("c?ol_1370")), with_166 (col_1371) AS (SELECT * FROM (VALUES (())) AS "ta
b_638" (col_1371)) SELECT "ct'e_ref_55".col̝_1369 AS "c!\fol_1372", decl_93 AS col_1373, true AS col_1374, "dec""l_94" AS col_1375, 1234666203:::REGPROCEDURE AS col_1376, "dec%ql_92" AS col_1377, 'BOX(-0.9835518624568584 -1.4364328427361834,-0.05228456231009149 0.37483813080537187)':::BOX2D AS "\\u06F6cOl_1378", "decl _98" AS col_1379, decl_95 AS col_1380, decl_99 AS "co)l_1381", 😲decl_88 AS "co̙l%v_1382" FROM "w ith_164" AS "ct'e_ref_55" WHERE false;
        NULL;
        WITH w😜ith_167 (col_1383) AS (SELECT * FROM (VALUES (e'\f':::STRING), (NULL), (bpchar(NULL::TSVECTOR)::BPCHAR), (bpchar(NULL::PG_LSN)::BPCHAR), ('t':::STRING)) AS tàb_639 (col_1383) INTERSECT ALL SELECT * FROM (VALUES (e'\x18':::STRING)) AS tab_640 ("
        col_1384")), with_168 (col_1385, "col	_1386") AS (SELECT * FROM (VALUES (1653034106:::REGCLASS, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET), (2040161004:::REGCLASS, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET), (NULL, '190.85.232.2/31':::INET)) AS "tab%v_641" (col_1385, "col	_1386")) SELECT false AS cól_1387 FROM with_168 AS cte_ref_56 ORDER BY cte_ref_56.col_1385 DESC NULLS LAST LIMIT 4:::INT8 INTO decl_97;
        SELECT regprocedure((-8335419415536647810):::INT8::INT8)::REGPROCEDURE AS col_1390 FROM (VALUES ('1977-04-04 12:08:51.000757+00':::TIMESTAMPTZ), ('2024-10-09 11:13:23.000026+00':::TIMESTAMPTZ), ('2026-02-09 11:39:36.000386+00':::TIMESTAMPTZ), ('1975-10-22 22:17:08.000192+00':::TIMESTAMPTZ), (NULL), (timezone(decl_100::STRING, parse_timestamp(e'\x0b"\'q\x1eOo':::STRING::STRING)::TIMESTAMP::TIMESTAMP)::TIMESTAMPTZ)) AS "tab&_642" ("co%ql_1388"), (VALUES (decl_90), (NULL), ("decl _98"), (decl_90), (decl_99)) AS "%vta b_643" ("%04col_1389") WHERE false LIMIT 38:::INT8 INTO "decl _98";
        END;
        $funcbodyxxxx$;
        FOR loop_62 IN 84053221795705613:::INT8..(-1):::INT8 BY (-32768):::INT8 LOOP
        decl_91 := B'10011010100010101111010';
        END LOOP;
        NULL;
        decl_95 := decl_95;
        NULL;
        NULL;
        NULL;
        END;
        $funcbodyxxxxx$;: at or near ";": at or near "decl_100": syntax error
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestGenerateParse2613300283
--- FAIL: TestGenerateParse (0.85s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGenerateParse.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-55315

internal/sqlsmith.TestGenerateParse [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20628270?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20628270?buildTab=artifacts#/) on master @ [a7bdedf4d7f04943a955ad86bcd2e9f7a9593b10](https://github.com/cockroachdb/cockroach/commits/a7bdedf4d7f04943a955ad86bcd2e9f7a9593b10):


```
        	FOR loop_28 IN REVERSE 1::INT8..5::INT8 LOOP
        END LOOP;
        ELSIF ('J'::STRING::STRING::STRING % 'G'::STRING::STRING::STRING)::BOOL THEN
        	"decl%v_78" := "decl%v_78";
        ELSIF true::BOOL THEN
        	SELECT "T{ab_599"."c
        ol_881" AS col_882, "decl%v_78" AS "co
l%a7_883" FROM (VALUES (NULL::TIMESTAMPTZ[])) AS "T{ab_599" ("c
        ol_881") WHERE NULL::BOOL ORDER BY "T{ab_599"."c
        ol_881" NULLS LAST;
        ELSE
        	"decl%v_78" := "decl%v_78";
        	SELECT "decl%v_78" AS col_885, ta😣b_600.col_884 AS " co/l_886" FROM (VALUES ('D'::STRING::VARCHAR), ('D'::STRING::VARCHAR)) AS ta😣b_600 (col_884) WHERE false::BOOL ORDER BY ta😣b_600.col_884 ASC NULLS FIRST, ta😣b_600.col_884 ASC NULLS FIRST, ta😣b_600.col_884 DESC;
        END IF;
        ELSIF false::BOOL THEN
        ELSIF NULL::BOOL THEN
        	"decl%v_78" := "decl%v_78";
        	SELECT 😮tab_601."c ol_887" AS "co l _888", "decl%v_78" AS "co\\x53l_889" FROM (VALUES ('J'::STRING::VARCHAR), ('B'::STRING::VARCHAR), ('G'::STRING::VARCHAR), (NULL::VARCHAR)) AS 😮tab_601 ("c ol_887") WHERE age('2000-01-01 00:00:00+00'::TIMESTAMPTZ::TIMESTAMPTZ::TIMESTAMPTZ, '2000-01-01 01:00:00+00'::TIMESTAMPTZ::TIMESTAMPTZ::TIMESTAMPTZ)::INTERVAL::INTERVAL IN ('00:00:03'::INTERVAL::INTERVAL,);
        	"decl%v_78" := "decl%v_78";
        ELSIF false::BOOL THEN
        	SELECT "decl%v_78" AS "c�o�l_890";
        ELSE
        	IF true::BOOL THEN
        	SELECT "decl%v_78" AS col_891;
        	SELECT "t\\uBB7Bab_602"."co%vl_892" AS co😂l_893 FROM (VALUES (NULL::GEOGRAPHY), (st_setsrid(st_intersection(NULL::GEOGRAPHY::GEOGRAPHY, NULL::GEOGRAPHY::GEOGRAPHY)::GEOGRAPHY::GEOGRAPHY, int4(NULL::CITEXT::CITEXT)::INT4::INT4)::GEOGRAPHY), (COALESCE(NULL::GEOGRAPHY, NULL::GEOGRAPHY))) AS "t\\uBB7Bab_602" ("co%vl_892") ORDER BY "t\\uBB7Bab_602"."co%vl_892" ASC NULLS LAST;
        	IF false::BOOL THEN
        	IF false::BOOL THEN
        	SELECT "decl%v_78" AS "col _903" FROM (VALUES (2::DECIMAL)) AS tab_603 ("col 😡_894"), (VALUES (NULL::BIT(37)), (B'0001'::BIT(37)), (B'0000'::BIT(37)), (B'0101'::BIT(37)), (B'0010'::BIT(37)), (B'0111'::BIT(37))) AS 😐tab_608 ("col""_899") JOIN (VALUES ('G'::STRING::STRING, '''KXeQPuol'' <-> ''SKpNBxP'' & ''Cr'''::TSQUERY::TSQUERY), ('J'::STRING::STRING, NULL::TSQUERY), ('C'::STRING::STRING, '''jtBTTeqc'' | ''vBQypyZ'' & ''hM'' | !''UMcweROw'''::TSQUERY::TSQUERY), ('I'::STRING::STRING, '''tURNFI'' | ''vJzayzzsI'' <-> !''ySTdw'' & ''cxAJ'''::TSQUERY::TSQUERY), ('C'::STRING::STRING, '!''gwfGR'''::TSQUERY::TSQUERY)) AS 😳tab_609 (co😖l_900, "coL _901") ON true::BOOL, (VALUES ('J'::STRING::"char"), ('C'::STRING::"char"), ('G'::STRING::"char"), ('B'::STRING::"char"), ('C'::STRING::"char")) AS "ta
        b	_610" ("co\vl_902") WHERE false::BOOL ORDER BY 😐tab_608."col""_899" ASC NULLS FIRST, 😐tab_608."col""_899" ASC INTO "decl%v_78";
        	"decl%v_78" := "decl%v_78";
        ELSIF false::BOOL THEN
        ELSIF NULL::BOOL THEN
        	SELECT tab_615.col_909 AS col_910 FROM (VALUES (NULL::RECORD), (NULL::RECORD), (NULL::RECORD), (NULL::RECORD), (NULL::RECORD), (NULL::RECORD)) AS tab_615 (col_909) GROUP BY tab_615.col_909 HAVING bool_or(NULL::BOOL::BOOL)::BOOL;
        ELSIF true::BOOL THEN
        	NULL;
        	SELECT "decl%v_78" AS col_911;
        END IF;
        	FOR loop_29 IN 0::INT8..NULL::INT8 LOOP
        SELECT ta͊b_616.col_912 AS "co l_913", "decl%v_78" AS col_914, loop_29 AS "co%vl_915" FROM (VALUES (NULL::REFCURSOR), (NULL::REFCURSOR), (NULL::REFCURSOR)) AS ta͊b_616 (col_912);
        END LOOP;
        END IF;
        END IF;
        END IF;
        NULL;
        END;
        $$;
        END;
        $funcbody$;: at or near "I": syntax error
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/0b91c216d93b1725b0e559055d162219/logTestGenerateParse2908983733
--- FAIL: TestGenerateParse (1.01s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGenerateParse.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

internal/sqlsmith.TestGenerateParse [failed](https://mesolite.cluster.engflow.com/invocations/default/53d496f4-c7c3-442f-bfef-a107ddadbab3?testReportRun=2&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvaW50ZXJuYWwvc3Fsc21pdGg6c3Fsc21pdGhfdGVzdA==) on master @ [901677f2dee53449e53708ff0aa318d8ee235c02](https://github.com/cockroachdb/cockroach/commits/901677f2dee53449e53708ff0aa318d8ee235c02):


```
        WITH "with""_209" (col_1393) AS (SELECT * FROM (VALUES ('=':::STRING), ('[':::STRING), (e'\U00002603':::STRING), ('t':::STRING), (NULL), (e'\'':::STRING)) AS "ta\\u7F81b_737" (col_1393)), 😶with_210 (col_1394) AS (SELECT * FROM (VALUES (B'10000010100001100101011011010100111101'), (B'11010101000000001100111001000101110101'), (B'00010110111011101100000111000110111101'), (NULL), (B'00000000000000000000000000000000000000'), (B'00111000010101110011000001011101000101')) AS "tab _738" (col_1394)), with_211 ("col%q_1400") AS (SELECT 466685027:::REGPROCEDURE AS "col%q_1400" FROM "with""_209" AS "c'te_😥reF_68", (VALUES ('10:02:41.938215+04:20':::TIMETZ, COALESCE(pg_lsn(NULL::PG_LSN)::PG_LSN, '5A4879B/8568A87D':::PG_LSN)), ('22:25:54.553531+07:15':::TIMETZ, 'A4013E49/3665C001':::PG_LSN), (timezone(st_ashexwkb('0103000020E6100000020000000500000000000000000000000000000000000000000000000000F03F0000000000000000000000000000F03F000000000000F03F0000000000000000000000000000F03F00000000000000000000000000000000050000009A9999999999C93F9A9999999999C93F9A9999999999C93F9A9999999999D93F9A9999999999D93F9A9999999999D93F9A9999999999D93F9A9999999999C93F9A9999999999C93F9A9999999999C93F':::GEOGRAPHY::GEOGRAPHY)::STRING::STRING, NULL::TIMETZ)::TIMETZ, 'E59FD343/3C2AB7C5':::PG_LSN), (('01:16:33.66958-12:26':::TIMETZ::TIMETZ + NULL::INTERVAL)::TIMETZ, NULL), ((SELECT timezone('':::STRING::STRING, '07:27:06.308925':::TIME::TIME)::TIMETZ AS col_1397 FROM (VALUES (regproc(e'\x05eg\x069Mxq~':::STRING::STRING)::REGPROC), (2833060620:::REGPROC), (regproc(e'\U00094E5A\U0001BFF3\U000B9E92\U00011495\U0006D00A' COLLATE und_u_ks_level2::CITEXT)::REGPROC), (840766250:::REGPROC), (1109450691:::REGPROC)) AS tab_739 (c😈ol_1395), (VALUES (NULL)) AS "t|ab_740" ("cOl_1396") ORDER BY "t|ab_740"."cOl_1396" ASC NULLS LAST, tab_739.c😈ol_1395 ASC, tab_739.c😈ol_1395 DESC NULLS LAST LIMIT 1:::INT8), 'F4CEBF0C/A69421BA':::PG_LSN), ('21:56:51.420163+06:06':::TIMETZ, 'CA325BB1/AAC8C5AC':::PG_LSN)) AS tab😀_741 (col_1398, coḻ_1399) WHERE true) SELECT jsonb_set(NULL::JSONB, ARRAY['':::STRING]::STRING[], "\\u0ACFdecl%v_161"::JSONB, true::BOOL)::JSONB AS col_1401 FROM 😶with_210 AS c̗te_ref_69 WHERE false LIMIT 87:::INT8;
        END;
        $$;
        NULL;
        END LOOP;
        	SELECT B'010101011011101100011011110110110101011110101001' AS co_l_1414, NULL AS "col	_1415", tab_742."\\U000DA497cOl_1402" AS "{co%ql%q�_1416", '00:00:00':::INTERVAL AS col_1417, '73 years 5 mons 389 days 04:34:50.916403':::INTERVAL AS "c̓ol%p_1418", '23:03:01.945368-12:36':::TIMETZ AS col_1419 FROM (VALUES (('fX4YR_w':::STRING:::NAME,)), ((NULL:::NAME,)), ((e'\x16':::STRING:::NAME,)), ((e'b\tmf&\x11':::STRING:::NAME,))) AS tab_742 ("\\U000DA497cOl_1402"), (VALUES (B'001111000000001011010101111011110110000101101000', '32 years 4 mons 12 days 20:48:40.171388':::INTERVAL, '1 day':::INTERVAL, (-7024381925819727038):::INT8), (B'000100011100010111110011011000001111110111111000', '-58 years -7 mons -685 days -14:58:12.701678':::INTERVAL, (WITH "with'_212" (col_1404) AS (SELECT '{"RzPrDF": false, "ZZ": true, "xcQE": {"CNwedTqCz": 0.050805995257900693, "bar": 1.8487096467885733, "hsP7VJOmF": "ELK5bQ0gl"}}':::JSONB AS col_1404 FROM (VALUES ('01050000400700000001020000400400000050E77D60FBA1F4417A49C73CE289F741002A481A09D3A24160497BD22060B241DE58FE249A8DF24106110672CFC3E0C1E00E81D21321EEC1F8E4635DCD8DE8410154C3EA1F1CF5C131F95CE94960F2C1049AB52CAB53E2412086639A3B35C8C1010200004008000000B4EDEEDACD8CD5C1162EBB8712DCF6C102692CD2C12DFDC17E5811E9347BFA416A4FC014BEBFF5C1164E7917BAF4F441FCB50F7B40970142424C33924D46E0C10B11B0A7C42301C2DE86FD8D3BD6F34180981DC58EE6B541A01A0D4E6A03F8C10C5342FC4A46F541D04521AA8F7AE941EAFBC0C28E5EE7C160FBC8DA036BEB4104566F382AA9E4413CF6216009A8FB41B0CDE4623AAFC5C1F08B9C35EDBBF04198ABEF77AA28FEC10632C9B20D29E4C18830F2203A34F54112461166DA3A00C20102000040020000000006AB9ECBE9DB41C4A3D5355FFDE54189803132458301C2E0E833760A7CB6417C1D3DA736E50042A80DE84885ADD6C101020000400200000086E6D0978075F041861BFC7CA6F0FDC1D506C1C40F20F6C14C4ECA0F1A0FEC41DEED26E6929DFA4177A0AFC9C0ADFBC1010200004002000000488D2D3154ABE641843B46358B9AFAC11BC72C38419FFDC11262B231212EF541F7240C7F4D3401C26C1A9D6C0AD8FCC1010200004003000000DE2ABF72973EE7C1DE71FD9F70E6EFC11C0C86F64121F3C1A0892EFFE2CFD5C17212FE7D3283FFC13839653EB123FDC106AE6B4BB36601428BE6A4D6330302C23BC86493D1F1F8C1010200004002000000045AF4BC821CF0C1B09341EF9BE1D5C1005C873830136441EA8B71986C8AEDC160865B4584A1C2416CA6D1BB418CFC41':::GEOMETRY), ('0104000080020000000101000080466FFDBEF22F02426694F96D61AE0042023ED2D8160C01C201010000802830FC7E8F8DD74192F4C26B9867E3C180FD9E4A4C75AF41':::GEOMETRY), ('010600000000000000':::GEOMETRY), ('01020000C003000000B05859748D5FD4C11800D24ABC3CFEC1DAD20A5715B5E6C18004301BD687B2C1E6BD09D0B7F5014238FB8917908CFD41487865EB8DBAF4C13E123565CF2CFDC140B3D25B7EC6F541F2E3EED1E68AF24108FE7A74C004E5C1084C4F3CA89FFC41':::GEOMETRY)) AS tab̮_743 (col_1403) WHERE true), "%61with_213" (col_1405) AS (SELECT * FROM (VALUES (('':::VOID, e'\U0007A8B9\U000913F6\U0007A37C' COLLATE da_DK, ((-2127995093):::INT8, (-1):::INT8))), (('':::VOID, e'\U00013F9A\U000D66FD' COLLATE da_DK, ((-838237619):::INT8, (-24744):::INT8))), (NULL), (('':::VOID, e'\U000A3B1E\U000258AA\U000621E7\U000E34E8\U000B9B77' COLLATE da_DK, ((-303311028):::INT8, NULL:::INT2))), (('':::VOID, e'\U0006C61D\U0001DB70' COLLATE da_DK, (1037025324:::INT8, 6030:::INT8))), (NULL)) AS "ta�b_744" (col_1405)), with_214 (col_1406, "c{ol _1407") AS (SELECT * FROM (VALUES (NULL, date_trunc(NULL::STRING, NULL::INTERVAL)::INTERVAL), ('''EDA'' ''MxYWklBMM'':164A,188B ''ODHpfjQEkY'':319C,919B ''cqsYoqQ'' ''uSToIimK'' ''usIZpGL''':::TSVECTOR, NULL), ('''AVlz'':336C,365B,629A ''KnN'' ''SWshMVTJ'' ''VLzgAcyrf'':584A,669B ''gqbwpE'':561C,799B ''lY'' ''qNj'':694C,697B ''ucn''':::TSVECTOR, '71 years 9 mons 334 days 22:03:51.915364':::INTERVAL), ('''DiQTcy'':346A,601C,866B ''LrPfvaL'':940C ''VIncnptlvo'':28A,140C ''ZLxF'' ''mZKsD'':108B,207A,363A,688C ''mqQPUmVh'':357A,551A,783C ''qUS'':423C,929A ''saQwdrbdE'' ''ynb'':322C,587B,643B,808A ''zAT''':::TSVECTOR, '-75 years -9 mons -376 days -15:03:14.725239':::INTERVAL)) AS " tab_745" (col_1406, "c{ol _1407")) SELECT '-53 years -8 mons -808 days -00:23:08.43837':::INTERVAL AS col_1408 FROM "%61with_213" AS "cte_re
f_70" WHERE false ORDER BY "cte_re
f_70".col_1405 DESC NULLS FIRST LIMIT 1:::INT8), 4802902445636713822:::INT8), (B'010111111111011001011110110111000010110111101100', '-3 years -4 mons -331 days -09:11:11.713605':::INTERVAL, '26 years 2 mons 966 days 20:09:20.115839':::INTERVAL, array_position(ARRAY['2009-05-27 21:43:30.000858+00':::TIMESTAMPTZ]::TIMESTAMPTZ[], '294276-12-31 23:59:59.999999+00':::TIMESTAMPTZ::TIMESTAMPTZ)::INT8), (B'011111111111111111111111111111111111111111111111', '70 years 3 mons 752 days 08:48:22.452907':::INTERVAL, '52 years 11 mons 739 days 13:52:46.378934':::INTERVAL, (-3577954140599604543):::INT8), (B'100111001111101100111001000011111011111000001100', NULL, parse_interval(e'\x04fMtQ<':::STRING::STRING, '':::STRING::STRING)::INTERVAL, 3555506224748289205:::INT8)) AS tab_746 (col_1409, col_1410, " Co""l_1411", " col_1412"), (VALUES ('08:08:23.421508+08':::TIMETZ), (COALESCE(('23:05:34.73735+09:08':::TIMETZ::TIMETZ - NULL::INTERVAL)::TIMETZ, NULL)), ('02:13:22.839768+10:41':::TIMETZ), ('00:00:00+15:59':::TIMETZ)) AS "t%va&b'_747" (co😮l😦_1413) WHERE false ORDER BY tab_746.col_1409 NULLS FIRST LIMIT 96:::INT8;
        	NULL;
        END IF;
        	DO $$
        DECLARE
        "dec
        l_168" GEOGRAPHY := '0107000020E6100000050000000105000020E6100000050000000102000000020000007838F162D0055B40C0C691B98AFD0CC02C4DEDBC1AEC59402C24FF5F19564440010200000002000000B8A3CEBD742E5440C01C77B0E181F2BF443565D420D65440F0DC4A2D882D2440010200000005000000F07341BD8AA1454050DB106653B54DC0ACDD3E5FF3B05640DBC84ED6313C50C084F32FF543F45840873EDCB3950841C0E6543042835462404407AA566E005640982B6B5F3D033F40E0ECCB171D113F40010200000004000000D30F0635F45D5AC0C44981F85CFA55C00005E1424CB75B404CB699CC713B4040C84F98C8334D5F408469B94B59084D40F8C8E5F61D1333C048A66B226C7550400102000000020000009A540E0DCFB5504000443BAC91B7414030AF5890AEB94840F87C56FD567148400105000020E610000004000000010200000002000000000447188B092340D1418CED66F153C0688D908A132C584026D117E344D94EC0010200000002000000629E71D0ACCF6040DB1E2AA5F71451C080B186F5BC2D6440D8EF9499BA3344C0010200000002000000208FA5388FC43AC0B4D4DDD2BA964F40789D5960CA6F49C0E8E3ECFBDF34504001020000000200000080B72EFD44C15DC03E3072A5D70E55401680029B2FC84FC0867833A3FB8341400106000020E61000000700000001030000000100000009000000AA8B47A44D2D55C00BF14B151B4C48C0055040C58BBC64C0088693B98FBA4EC080A98F10B581F6BFA4C0AE59946F50C01CF76FC0F33662400CF14D0901494EC0E81DC9F253195C40DD84AA8A5AE64AC0A06C719F8E665540FAC2AB45A54245C022D4CB43CBBE6540F00D9921B7E71540C68CB120ADA562C068ACFEC432964240AA8B47A44D2D55C00BF14B151B4C48C00103000000010000000D0000002421D1ACE9134CC0E0B66C0742D30740D13E1AAA598D61C034C26491B24240C0AA9EBB1F1C7B5DC069FB3C6A0E7548C0A0DAC504F92044C0582B0ECE517D4DC0C26E4620A0D2644026416C98AB0C4FC0BE15EF6BC49D61405060E3BF321711C0701B724F9F89654000FE7A2E7A4F364080F647DBE30BFABF5CDA4A9E6AFC464068F6EAB7992F40C08CFA7548C11E5340F08DD27C9A1B3BC0B0EEE14123E5464041CF07D2AA6065C0F0B72F6D51C44D40AC750A7CEBD660C054E4DDD12EB544402421D1ACE9134CC0E0B66C0742D307400103000000010000000D00000032DCD1CD012063C0540E58BD7B9C42C0B308F34006E061C0C4BB84D86F2E52C090A85F21DE6232C06408CDADEC0A53C0AEDEB2B93C436440BFF83462734252C0783B4FE1C65C54406024928FBB310F40DAEF3A60EA7A6340CCB3BD6723F53B40046C2E903DE24540724F56468B9054404097F2E383490C40D46CDFFBD6A042400CFAB5E1EF4043C06CAACDD8B2524A40656C75B21BD161C05AB8DFB524105640071D7981C38F5DC0309BB154A4334940160D93F8085B55C078F0AAD4F217224032DCD1CD012063C0540E58BD7B9C42C0010300000001000000060000008385AF27233D65C02ABF09E3394B55C0FDF9D03D42D255C0209945EBC99552C044737D6CC5C55540F851352BFB854AC0FC6575A6FB2A5440C01E1B4D6EF0494097F0BB43E65060C09881058EAE3350408385AF27233D65C02ABF09E3394B55C0010300000001000000070000004C14994D7E765640E043428ABF694DC01C94368105A860406C7258E5824E52406297C814DD086040DECC261015065640888616A092753A40CC2198067D0154404A56ADF3409865C080C9E45D05014D40C45C731437C45DC0843645E4AD724A404C14994D7E765640E043428ABF694DC00103000000010000000C00000030819B5AD90962C09E3F57714CE349C0145762C17AD465C09539D5E3062552C0BE5622078DCC57C0AC22F4C54E2145C0BC9477D3B3DB4A40B2408A9E165B4DC040CED4245AC04540441CBDC1167B3CC06E73471955BD6440545B5D4B0FC44B4068EB7549D60236C098555E7BC9413F402EBB6EF1019A62C044D827F648434540FFCDA9ECBC2066C068F9AF10265A4C40471214F46EF360C0D02435090A7F134020BDAADF8C0257C0D0313BA8BA0215C030819B5AD90962C09E3F57714CE349C001030000000100000005000000B01B9BB126473A4080CEBF12259E2F40FA3CB1934305604058EFD4F870892AC0F03C423D268A624098BE28EBA13E5640780AB78A5D6A36C0C477E1A28DAF4940B01B9BB126473A4080CEBF12259E2F400104000020E61000000100000001010000005475CDED87C14DC0AA14FAD7BF213AC00101000020E6100000A83A43E25CB065C04031BFB1E4D24740':::GEOGRAPHY;
        decl_169 REFCURSOR;
        decl_170 CONSTANT NAME := e'(\x1f':::STRING:::NAME;
        decl_171 TIMESTAMPTZ;
        "decl%c0_172" NAME;
        decl͎_173 TIMESTAMP;
        decl_174 CONSTANT TIMESTAMP := NULL;
        BEGIN
        decl͎_173 := NULL;
        NULL;
        decl_171 := decl_171;
        FOR "loo p_41" IN REVERSE (-2876823447671236496):::INT8..(-3897525428636897049):::INT8 BY (-8669273488421269257):::INT8 LOOP
        END LOOP;
        END;
        $$;
        	SELECT 3057925805:::REGROLE AS col_1421 FROM (VALUES (4227023546:::REGROLE), (546769054:::REGROLE)) AS tab_748 ("😶c
        ol_1420") WHERE true LIMIT 86:::INT8;
        ELSIF true THEN
        	NULL;
        	NULL;
        ELSIF true THEN
        	NULL;
        	NULL;
        	NULL;
        ELSIF false THEN
        	WITH "with�_215" (col_1422) AS (SELECT * FROM (VALUES ('LqFF6mef.ucV_AYL.iAWOWFHc.BrJFs9UMv.M'), (NULL), ('rXvlr'), ('ftgrwq_m.ohpdz.EOIe.O3X')) AS tab_749 (col_1422)) SELECT tab_752.col_1425 AS ")c ol_1426" FROM (VALUES (NULL)) AS tab_752 (col_1425) ORDER BY tab_752.col_1425 ASC NULLS FIRST LIMIT 17:::INT8;
        	SELECT 8775920053060421636:::INT8 AS col_1428 FROM (VALUES ((-5923026441315449580):::INT8), (((-5034180418924837888):::INT8::INT8 - 2913024244209335664:::INT8::INT8)::INT8), ((-8392792182876415954):::INT8), (5100935304439857827:::INT8), (NULL)) AS tab_753 (" col_1427") ORDER BY tab_753." col_1427" NULLS FIRST, tab_753." col_1427" ASC, tab_753." col_1427" ASC NULLS LAST LIMIT 7:::INT8;
        	WHILE '4714-11-24 BC':::DATE::DATE IN ('1993-07-26':::DATE, '4714-11-24 BC':::DATE, '1986-05-14':::DATE, '1994-02-03':::DATE) LOOP
        WHILE '\x174e':::BYTES::BYTES IN (WITH "with\\U0007C69B_216" (col͟_1429) AS (SELECT * FROM (VALUES ('''DLbSrxmuyi'':92A,192B,550A ''G'' ''N'':14B,251A,383C ''PGPkA'':80A,686C ''XaJyzZuRs'':943A ''YUlcTOgtq'':16C,652B,933B':::TSVECTOR)) AS "tab
_754" (col͟_1429)), "wi""th_217" (col_1430) AS (SELECT * FROM (VALUES ((-0.8859375115129551):::FLOAT8), (((-1.2108776297045958):::FLOAT8::FLOAT8 % (-1.0):::FLOAT8::FLOAT8)::FLOAT8), (NULL)) AS "tab�😈_755" (col_1430)), "%w%vith_218" ("""col}_1431") AS (SELECT * FROM (VALUES (e'\x1e':::STRING), ('u':::STRING)) AS "tab _756" ("""col}_1431")) SELECT '\x2e57':::BYTES::BYTES AS col_1433 FROM (VALUES (NULL), (ARRAY[]:::BOOL[]), (ARRAY[false,false,true])) AS tab_757 (col_1432) WHERE true LIMIT 88:::INT8) LOOP
        END LOOP;
        SELECT (-8506207464457762321):::INT8 AS "?co.lͬ_1434" LIMIT 48:::INT8;
        NULL;
        END LOOP;
        END IF;
        END;
        $funcbodyx$;: at or near "�J
gm": syntax error
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestGenerateParse124034895
--- FAIL: TestGenerateParse (5.41s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGenerateParse.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/internal/sqlsmith/plpgsql.go
+++ b/pkg/internal/sqlsmith/plpgsql.go
@@ -14,6 +14,8 @@ import (
 	"github.com/cockroachdb/errors"
 )
 
+var plpgsqlFlags = tree.FmtParsable | tree.FmtPLpgSQLParen
+
 func (s *Smither) makeRoutineBodyPLpgSQL(
 	params tree.ParamTypes, rTyp *types.T, vol tree.RoutineVolatility,
 ) string {
@@ -25,7 +27,7 @@ func (s *Smither) makeRoutineBodyPLpgSQL(
 	// errors.
 	block := s.makePLpgSQLBlock(scope)
 	block.Body = append(block.Body, s.makePLpgSQLReturn(scope))
-	return "\n" + tree.AsStringWithFlags(s.makePLpgSQLBlock(scope), tree.FmtParsable)
+	return "\n" + tree.AsStringWithFlags(s.makePLpgSQLBlock(scope), plpgsqlFlags, tree.FmtInPLpgSQL(true /* inPLpgSQL */))
 }
 
 func (s *Smither) makePLpgSQLBlock(scope plpgsqlBlockScope) *ast.Block {

--- a/pkg/internal/sqlsmith/setup_test.go
+++ b/pkg/internal/sqlsmith/setup_test.go
@@ -8,8 +8,6 @@ package sqlsmith_test
 import (
 	"context"
 	"flag"
-	"fmt"
-	"strings"
 	"testing"
 
 	"github.com/cockroachdb/cockroach/pkg/base"
@@ -92,9 +90,7 @@ func TestGenerateParse(t *testing.T) {
 		t.Fatalf("unknown setting %s", settingName)
 	}
 	settings := setting(rnd)
-	t.Log("setting:", settingName, settings.Options)
 	setupSQL := setup(rnd)
-	t.Log(strings.Join(setupSQL, "\n"))
 	for _, stmt := range setupSQL {
 		db.Exec(t, stmt)
 	}
@@ -119,7 +115,6 @@ func TestGenerateParse(t *testing.T) {
 		if err != nil {
 			t.Fatal(err)
 		}
-		fmt.Print("STMT: ", i, "\n", stmt, ";\n\n")
 		if *flagExec {
 			_, err = conn.ExecContext(ctx, `SET statement_timeout = '9s'`)
 			if err != nil {
@@ -129,7 +124,7 @@ func TestGenerateParse(t *testing.T) {
 				es := err.Error()
 				if !seen[es] {
 					seen[es] = true
-					fmt.Printf("ERR (%d): %v\n", i, err)
+					t.Logf("ERR (%d): %v\n", i, err)
 				}
 			}
 		}

--- a/pkg/internal/sqlsmith/sqlsmith.go
+++ b/pkg/internal/sqlsmith/sqlsmith.go
@@ -192,6 +192,7 @@ var prettyCfg = func() tree.PrettyCfg {
 	cfg := tree.DefaultPrettyCfg()
 	cfg.LineWidth = 120
 	cfg.Simplify = false
+	cfg.FmtFlags = tree.FmtPLpgSQLParen
 	return cfg
 }()
 
@@ -213,10 +214,10 @@ func (s *Smither) Generate() string {
 		i = 0
 
 		printCfg := prettyCfg
-		fl := tree.FmtParsable
+		fl := plpgsqlFlags
 		if s.postgres {
-			printCfg.FmtFlags = tree.FmtPGCatalog
-			fl = tree.FmtPGCatalog
+			printCfg.FmtFlags = tree.FmtPGCatalog | tree.FmtPLpgSQLParen
+			fl = tree.FmtPGCatalog | tree.FmtPLpgSQLParen
 		}
 		p, err := printCfg.Pretty(stmt)
 		if err != nil {

--- a/pkg/sql/sem/plpgsqltree/statements.go
+++ b/pkg/sql/sem/plpgsqltree/statements.go
@@ -1121,6 +1121,7 @@ func (s *DoBlock) Format(ctx *tree.FmtCtx) {
 		// Format the body of the DO block separately so that FormatStringDollarQuotes
 		// can examine the resulting string and determine how to quote the block.
 		bodyCtx := ctx.Clone()
+		tree.FmtInPLpgSQL(true /* inPLpgSQL */)(bodyCtx)
 		bodyCtx.FormatNode(s.Block)
 		bodyStr := "\n" + bodyCtx.CloseAndGetString()
 

--- a/pkg/sql/sem/tree/expr.go
+++ b/pkg/sql/sem/tree/expr.go
@@ -1438,6 +1438,13 @@ type CaseExpr struct {
 
 // Format implements the NodeFormatter interface.
 func (node *CaseExpr) Format(ctx *FmtCtx) {
+	if ctx.HasFlags(FmtPLpgSQLParen) && ctx.inPLpgSQL {
+		// In some cases in PLpgSQL context we need to wrap the CASE expression
+		// in parenthesis to make it parsable. We do so only if the caller
+		// requested it.
+		ctx.WriteByte('(')
+		defer ctx.WriteByte(')')
+	}
 	ctx.WriteString("CASE ")
 	if node.Expr != nil {
 		ctx.FormatNode(node.Expr)

--- a/pkg/sql/sem/tree/format.go
+++ b/pkg/sql/sem/tree/format.go
@@ -201,6 +201,10 @@ const (
 
 	// FmtHideHints skips over any hints.
 	FmtHideHints
+
+	// FmtPLpgSQLParen will wrap some expressions in parenthesis when in PLpgSQL
+	// context. This should only be used in tests.
+	FmtPLpgSQLParen
 )
 
 const genericArityIndicator = "__more__"
@@ -343,6 +347,11 @@ type FmtCtx struct {
 	// indexedTypeFormatter is an optional interceptor for formatting
 	// IDTypeReferences differently than normal.
 	indexedTypeFormatter func(*FmtCtx, *OIDTypeReference)
+
+	// inPLpgSQL, if set, indicates that we're formatting a node within PLpgSQL
+	// context.
+	inPLpgSQL bool
+
 	// small scratch buffer to reduce allocations.
 	scratch [64]byte
 }
@@ -404,6 +413,14 @@ func FmtLocation(loc *time.Location) FmtCtxOption {
 	}
 }
 
+// FmtInPLpgSQL modifies FmtCtx to indicate whether we're in the PLpgSQL
+// context.
+func FmtInPLpgSQL(inPLpgSQL bool) FmtCtxOption {
+	return func(ctx *FmtCtx) {
+		ctx.inPLpgSQL = inPLpgSQL
+	}
+}
+
 // NewFmtCtx creates a FmtCtx; only flags that don't require Annotations
 // can be used.
 func NewFmtCtx(f FmtFlags, opts ...FmtCtxOption) *FmtCtx {
@@ -423,14 +440,15 @@ func NewFmtCtx(f FmtFlags, opts ...FmtCtxOption) *FmtCtx {
 // original.
 func (ctx *FmtCtx) Clone() *FmtCtx {
 	newCtx := fmtCtxPool.Get().(*FmtCtx)
+	newCtx.dataConversionConfig = ctx.dataConversionConfig
+	newCtx.location = ctx.location
 	newCtx.flags = ctx.flags
 	newCtx.ann = ctx.ann
 	newCtx.indexedVarFormat = ctx.indexedVarFormat
 	newCtx.placeholderFormat = ctx.placeholderFormat
 	newCtx.tableNameFormatter = ctx.tableNameFormatter
 	newCtx.indexedTypeFormatter = ctx.indexedTypeFormatter
-	newCtx.dataConversionConfig = ctx.dataConversionConfig
-	newCtx.location = ctx.location
+	newCtx.inPLpgSQL = ctx.inPLpgSQL
 	return newCtx
 }
 

